var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -476,7 +476,15 @@ def determine_bootstrap_options(self, spec, with_libs, options):
         with open("user-config.jam", "w") as f:
             # Boost may end up using gcc even though clang+gfortran is set in
             # compilers.yaml. Make sure this does not happen:
-            f.write("using {0} : : {1} ;\n".format(boost_toolset_id, spack_cxx))
+            if not spec.satisfies("%intel") and not spec.satisfies("platform=windows"):
```

Comment:
Was this happening with just the classic intel compiler or with oneapi, as well?